cluster/images/conformance/run_e2e.sh

```diff
@@ -39,23 +39,28 @@ saveResults() {
 # We get the TERM from kubernetes and handle it gracefully
 trap shutdown TERM
 
-ginkgo_args=(
-    "--focus=${E2E_FOCUS}"
-    "--skip=${E2E_SKIP}"
-    "--noColor=true"
-)
-
+ginkgo_args=()
 if [[ -n ${E2E_DRYRUN:-} ]]; then
     ginkgo_args+=("--dryRun=true")
 fi
 
 case ${E2E_PARALLEL} in
-    'y'|'Y')           ginkgo_args+=("--nodes=25") ;;
-    [1-9]|[1-9][0-9]*) ginkgo_args+=("--nodes=${E2E_PARALLEL}") ;;
+    'y'|'Y'|'true')
+        # The flag '--p' will automatically detect the optimal number of ginkgo nodes.
+        ginkgo_args+=("--p")
+        # Skip serial tests if parallel mode is enabled.
+        E2E_SKIP="\\[Serial\\]|${E2E_SKIP}" ;;
 esac
 
-echo "/usr/local/bin/ginkgo ${ginkgo_args[@]} /usr/local/bin/e2e.test -- --disable-log-dump --repo-root=/kubernetes --provider=\"${E2E_PROVIDER}\" --report-dir=\"${RESULTS_DIR}\" --kubeconfig=\"${KUBECONFIG}\""
-/usr/local/bin/ginkgo "${ginkgo_args[@]}" /usr/local/bin/e2e.test -- --disable-log-dump --repo-root=/kubernetes --provider="${E2E_PROVIDER}" --report-dir="${RESULTS_DIR}" --kubeconfig="${KUBECONFIG}" | tee ${RESULTS_DIR}/e2e.log &
+ginkgo_args+=(
+    "--focus=${E2E_FOCUS}"
+    "--skip=${E2E_SKIP}"
+    "--noColor=true"
+)
+
+set -x
+/usr/local/bin/ginkgo "${ginkgo_args[@]}" /usr/local/bin/e2e.test -- --disable-log-dump --repo-root=/kubernetes --provider="${E2E_PROVIDER}" --report-dir="${RESULTS_DIR}" --kubeconfig="${KUBECONFIG}" | tee "${RESULTS_DIR}"/e2e.log &
```

Comment:
I suspect this newfangled way of constructing the `ginkgo` command line has consequences for how special characters in `E2E_SKIP` AND `E2E_FOCUS` need to be escaped since the same values for these variables that worked on older kubernetes versions (using the old heptio conformance image) cause my conformance tests to run 0 tests total. This combined with the changes in this PR to the way `E2E_PARALLEL` is handled was sufficient to fool me into thinking that my tests were passing since `sonobuoy e2e <tarball>` only shows the number of failed tests, nothing about the total number of tests run.